spec.bs

```diff
@@ -3474,6 +3477,10 @@ Each {{InterestGroupReportingScriptRunnerGlobalScope}} has a
 <div algorithm>
   The <dfn method for="InterestGroupReportingScriptRunnerGlobalScope">registerAdMacro(|name|,
   |value|)</dfn> method steps are:
+  1. [=list/For each=] [=code unit=] |c| in |name| and |value|:

```

Comment:
How can you have a single for each for two separate things, captured by the same character `c`? I'm not sure I understand what this is trying to do.

Reply:
I think it wanted to say "for each |c| in |name|, ...; for each |c| in |value|: do the same", or "let |nameAndValue| be a new list with |name| and |value|'s elements; for each |c| in |nameAndValue|".

Reply:
OK then please update accordingly? I still don't really understand though. The first alternative you provided works if name and value are the same length, but breaks in general because you're using `|c|` twice so I think it's just getting double-assigned / overwritten maybe? It is certainly not equivalent to the second alternative you provided, which straight up concatenates `|name|` and `|value|`, but I'm not sure why. The third is equally confusing to me. When would you choose `|name|` _or_ `|value|` to pull `|c|` from? Please make this less ambiguous.

Reply:
I updated this to just use 2 for loops, one for |name| and one for |value|, so that there is absolutely no ambiguity.

Not that it matters, but to clarify what I meant:

"For each |c| in |name| and for each |c| in |value|": "For each |c| in name" and "for each |c| in value" are each iterators. I think it's natural for "and" of iterators to mean "and then", but you could also view it as set intersection I guess.

Then "For each |c| in concat(|name|, |value|)" is just lowering the "and then" into the string level, rather than the iterator.

Then "For each |c| in |name| or |value|" means "For each |c| such that |c| is in |name| or |c| is in |value|", which would iterate a different number of times (no duplicates) but is functionally equivalent here.

Reply:
@domfarolino this seems to be addressed. Please see if this can be approved. Thanks!